src/shared/fork-journal.c

```diff
@@ -196,9 +196,12 @@ void journal_terminate(PidRef *pidref) {
                 return;
 
         r = pidref_kill(pidref, SIGTERM);
-        if (r < 0)
-                log_debug_errno(r, "Failed to send SIGTERM to journalctl child " PID_FMT ", ignoring: %m", pidref->pid);
+        if (r != -ESRCH) {
+                if (r < 0)
+                        log_debug_errno(r, "Failed to send SIGTERM to journalctl child " PID_FMT ", ignoring: %m", pidref->pid);
+
+                (void) pidref_wait_for_terminate_and_check("journalctl", pidref, /* flags= */ 0);
+        }
 
-        (void) pidref_wait_for_terminate_and_check("journalctl", pidref, /* flags= */ 0);
```

Comment:
hmm, we should reap the process regardless if SIGTERM worked or not. If the process is already dead it generally doesn't mean it's also reaped.